test/scenarios/ynLSD.spec.sol

```diff
@@ -1,78 +1,217 @@
+		previousTotalShares += user2Shares;
+
+		runInvariants(IERC20(asset), user3, previousTotalAssets, previousTotalDeposited, previousTotalShares, user3Amount, user3Shares);
+	}
+
+	function runInvariants(IERC20 asset, address user, uint256 previousTotalAssets, uint256 previousTotalDeposited, uint256 previousTotalShares, uint256 userAmount, uint256 userShares) public view {
+		Invariants.totalDepositIntegrity(ynEigenToken.assetBalance(asset), previousTotalDeposited, userAmount);
+		Invariants.totalAssetsIntegrity(ynEigenToken.totalAssets(), previousTotalAssets, assetRegistry.convertToUnitOfAccount(asset, userAmount));
+		Invariants.shareMintIntegrity(ynEigenToken.totalSupply(), previousTotalShares, userShares);
+		Invariants.userSharesIntegrity(ynEigenToken.balanceOf(user), 0, userShares);
+	}
+}
+
+contract YnLSDScenarioTest2 is ynEigenIntegrationBaseTest {
+
+	/**
+		Scenario 2: Deposit Paused 
+		Objective: Ensure that deposits are correctly 
+		paused and resumed, preventing or allowing ETH 
+		deposits accordingly.
+	 */
+
+	address user1 = address(0x01);
+	address user2 = address(0x02);
+
+	// pause ynLSD and try to deposit fail
+	function test_ynLSD_Scenario_2_Pause() public {
+
+		vm.prank(actors.ops.PAUSE_ADMIN);
+	 	ynEigenToken.pauseDeposits();
+
+		address asset = chainAddresses.lsd.WSTETH_ADDRESS;
+		deal({ token: asset, to: user1, give: 1 ether });
+		vm.startPrank(user1);
+		IERC20(asset).approve(address(ynEigenToken), 1 ether);
+		vm.expectRevert(bytes4(keccak256(abi.encodePacked("Paused()"))));
+		ynEigenToken.deposit(IERC20(asset), 1 ether, user1);
+		vm.stopPrank();
+	}
+
+	function test_ynLSD_Scenario_2_Unpause() public {
+
+	 	vm.prank(actors.ops.PAUSE_ADMIN);
+	 	ynEigenToken.pauseDeposits();
+		assertTrue(ynEigenToken.depositsPaused());
+	 	vm.prank(actors.admin.UNPAUSE_ADMIN);
+		ynEigenToken.unpauseDeposits();
+		assertFalse(ynEigenToken.depositsPaused());
+		vm.stopPrank();
+
+		address asset = chainAddresses.lsd.WSTETH_ADDRESS;
+		deal({ token: asset, to: user1, give: 1 ether });
+		vm.startPrank(user1);
+		IERC20(asset).approve(address(ynEigenToken), 1 ether);
+		ynEigenToken.deposit(IERC20(asset), 1 ether, user1);
+		vm.stopPrank();
+		assertEq(ynEigenToken.balanceOf(user1), assetRegistry.convertToUnitOfAccount(IERC20(asset), 1 ether));
+	}
+
+	function test_ynLSD_Scenario_2_Pause_Transfer(uint256 random1) public {
+		vm.assume(random1 > 0 && random1 < 100_000_000 ether);
+		
+		uint256 amount = random1;
+		address asset = chainAddresses.lsd.WSTETH_ADDRESS;
+		deal({ token: asset, to: user1, give: amount });
+		vm.startPrank(user1);
+		IERC20(asset).approve(address(ynEigenToken), amount);
+		ynEigenToken.deposit(IERC20(asset), amount, user1);
+		assertEq(ynEigenToken.balanceOf(user1), assetRegistry.convertToUnitOfAccount(IERC20(asset), amount));
+
+		// should fail when not on the pause whitelist
+		ynEigenToken.approve(user2, amount);
+		vm.expectRevert(bytes4(keccak256(abi.encodePacked("TransfersPaused()"))));
```

Comment:
please use expectRevert with TransfersPaused.selector as in the other tests for consistency

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/135/commits/e233d4b76e889c6ee9cfeb3ff481d842ad4b2df1